app/lib/common/utilities/pdf_preview.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_full_pdf_viewer/full_pdf_viewer_scaffold.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:pdf/pdf.dart';
+import 'package:pdf/widgets.dart' as pw;
+
+class PdfPreview extends StatefulWidget {
+  @override
+  _PdfPreviewState createState() => _PdfPreviewState();
+}
+
+class _PdfPreviewState extends State<PdfPreview> {
+  Future<File> _pdf;
+
+  @override
+  void initState() {
+    super.initState();
+    _pdf = _getBuiltAndSavedPdf();
+  }
+
+  @override
+  build(BuildContext context) {
+    return FutureBuilder<File>(
+        future: _pdf,
+        builder: (BuildContext context, AsyncSnapshot<File> snapshot) {
+          switch (snapshot.connectionState) {
+            case ConnectionState.waiting:
+              // TODO: Replace this with loading screen
+              return Text('Loading PDF');
+            default:
+              if (snapshot.hasError)
+                return Text('Could not build and save PDF: ${snapshot.error}');
+              else
+                return PDFViewerScaffold(
+                  appBar: AppBar(
+                    title: Text('Example PDF'),
+                  ),
+                  path: snapshot.data.path,
+                );
+          }
+        });
+  }
+
+  // TODO: Give this function an object to build PDF from
+  Future<File> _getBuiltAndSavedPdf() async {
```

Comment:
I think normally these would be separate functions, but I don't see why we'd ever build a PDF to then not save/use it.